### PR TITLE
feature: auto-complete paths for plugins and assets

### PR DESCRIPTION
**Pull request by EvanBacon**
Inspired by [path-intellisense](https://marketplace.visualstudio.com/items?itemName=christian-kohler.path-intellisense), this enables autocomplete for items in the `plugins` array, and for images matching a set amount of properties. 

When the activation is `""`, valid dependencies in the package.json `dependencies` object will be recommended (they'll be sorted according to likeliness for the package to have a plugin (i.e. has app.plugin.js entry file)). 
When the activation is `"./"`, JS files will be recommended. Regular files won't be validated before being recommended, we fallback on the existing validation after the resolver has been defined.

When one of the allowed image props is used, png, and jpeg files will be recommended:
```
  'xxhdpi',
  'xhdpi',
  'hdpi',
  'mdpi',
  'image',
  'tabletImage',
  'foregroundImage',
  'backgroundImage',
  'icon',
  'favicon',
  'xxxhdpi',
```

### CODE REVIEW

**Comment by EvanBacon** [comment]
Might be worth looking at this plugin as well https://github.com/leqwasd/VsCodeAsyncJsonAutocomplete/

File: src/completion/provider/configuration/getMapping.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import * as vscode from 'vscode';
+
+export interface Mapping {
+  key: string;
+  value: string;
+}
+/**
+ * From `{ "lib": "libraries", "other": "otherpath" }`
+ * To `[ { key: "lib", value: "libraries" }, { key: "other", value: "otherpath" } ]`
```
Comment (byCedric) [inline src/completion/provider/configuration/getMapping.ts:9]: 👍  this helps a lot

File: src/completion/provider/provideCompletionItems.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const config = await getConfiguration(document.uri);
+
+  return provideAsync(context, config);
+}
+
+/**
+ * Read the package.json and get all of the dependencies that have a valid config plugin.
+ *
+ * @param packageJsonPath
+ * @returns
```
Comment (byCedric) [inline src/completion/provider/provideCompletionItems.ts:30]: Maybe just rely on TS's inferred or explicit returns might be better than just `@returns`?

**Review by byCedric — APPROVED** [review]

**Comment by byCedric** [comment]
Looks good and works like a charm, pretty fun how the autocompletion removes any doubt of the supported `expo-` packages I can use as plugin!

**Pull request merged**